README.Rmd

```diff
@@ -57,130 +55,169 @@ library(sfheaders)
 ```{r}
 fold1 <- rbind(c(0, 0), c(5, 0), c(5, 5), c(0, 5), c(0, 0))
 fold2 <- rbind(c(5, 0), c(10, 0), c(10, 5), c(5, 5), c(5, 0))
-blocks <- bind_rows(
-  sf_polygon(fold1),
-  sf_polygon(fold2)
+
+blocks_ex <- bind_rows(
+  sfheaders::sf_polygon(fold1),
+  sfheaders::sf_polygon(fold2)
 ) %>%
-  mutate(foldID = c(1, 2))
+  mutate(folds = c(1, 2) %>% factor())
 ```
 
-Next we assign each point to folds using `blockCV::spatialBlock()`
+Next we assign each tree to the correct folds using the `foldID` variable of the output returned by the `spatialBlock()` function from the [`blockCV`](https://github.com/rvalavi/blockCV) package.
 
 ```{r}
-ex_cv_grid <- spatialBlock(
-  speciesData = ex_growth_df,
-  verbose = FALSE,
-  k = 2,
-  selection = "systematic",
-  blocks = blocks,
-  showBlocks = FALSE
+SpatialBlock_ex <- blockCV::spatialBlock(
+  speciesData = growth_ex, k = 2, selection = "systematic", blocks = blocks_ex,
+  showBlocks = FALSE, verbose = FALSE
 )
 
-# Add foldID to data
-ex_growth_df <- ex_growth_df %>%
-  mutate(foldID = ex_cv_grid$foldID %>% as.factor())
+growth_ex <- growth_ex %>%
+  mutate(foldID = SpatialBlock_ex$foldID %>% factor())
+```
 
-# Deliverable
-ggplot() +
-  geom_sf(data = ex_growth_df, aes(col = buffer, shape = foldID), size = 2) +
-  geom_sf(data = blocks, fill = "transparent")
+In the visualization below, the spatial blocks that act as our cross-validation folds are delineated in orange. Correspondingly the shape of each point indicates which fold each tree has been assigned to. 
 
-# Convert to sf object
-ex_cv_grid_sf <- ex_cv_grid$blocks %>%
-  st_as_sf()
+```{r}
+base_plot + 
+  geom_sf(data = growth_ex, aes(col = buffer, shape = foldID), size = 2) +
+  geom_sf(data = blocks_ex, fill = "transparent", col = "orange")
 ```
 
-We then create a `focal_v_comp` tibble which has a row for each competing pair of individuals (each pair of individuals within `max_dist` of one another):
+### Compute focal versus competitor tree information
+
+Based on our `growth` data frame, we now explicitly define all "focal" trees and their respective "competitor" trees in a `focal_vs_comp` data frame. This data frame has rows corresponding to each focal tree and all information about its competitors are saved in list-column variable `comp`. We implemented this nested format using `tidyr::nest()` in order to minimize redundancy, given that the same tree can act as a competitor multiple times. 
+
+```{r}
+focal_vs_comp_ex <- growth_ex %>%
+  create_focal_vs_comp(comp_dist, cv_grid_sf = blocks_ex, id = "ID")
+focal_vs_comp_ex
+```
+
+Using `unnest()` we can fully expand the competitor information saved in the `focal_vs_comp` data frame. For example the tree with `focal_ID` equal to 2 located at (1.5, 2.5) has two competitors within `comp_dist` distance from it.  
 
 ```{r}
-focal_vs_comp_ex <- ex_growth_df %>%
-  create_focal_vs_comp(max_dist, cv_grid_sf = ex_cv_grid_sf, id = "ID")
+focal_vs_comp_ex %>% 
+  tidyr::unnest(cols = "comp")
```

Comment:
Adding a `library(tidyr)` call so that the user knows what they need to install from reading the `library` chunk. :-)